### PR TITLE
Update doc for engine headers authentication

### PR DESCRIPTION
**Pull request by jdrossl**
https://github.com/craftercms/craftercms/issues/4603

### CODE REVIEW

File: source/site-administrators/engine/engine-headers-based-authentication.rst [hunk 1]
```diff
@@ -8,14 +8,18 @@
...
+ headers based authentication.
 
 To enable Engine headers based authentication:
 
 - Set ``security.headers.standalone`` to ``true``
 - Set the URLs requiring authentication
 
-Additionally, optional role mappings are available that allows mapping names from the external authentication to simple role names to use in the page or url restrictions.  Optional attribute mappings are also available which allows exposing attributes from the external authentication.
+Additionally, optional role mappings are available that allows mapping names from the external authentication to
+simple role names to use in the page or url restrictions.  Optional attribute mappings are also available which
```
Comment (sumerjabri) [inline source/site-administrators/engine/engine-headers-based-authentication.rst:21]: URL

File: source/site-administrators/engine/engine-headers-based-authentication.rst [hunk 2]
```diff
@@ -8,14 +8,18 @@
...
+allows exposing attributes from the external authentication.
```
Comment (sumerjabri) [inline source/site-administrators/engine/engine-headers-based-authentication.rst:22]: allow (not allows)

the external authentication authority.

File: source/site-administrators/engine/engine-headers-based-authentication.rst [hunk 3]
```diff
@@ -58,16 +63,50 @@ Next, configure the URLs you require authentication by setting ``url`` to desire
 
 See :ref:`engine-site-security-guide-restrict-urls` for more information on expressions that cam be used.
 
-From the above configuration, here are the attributes that Engine expects from the headers to be provided:
+From the above configuration, here are the headers that Engine expects to be provided:
 
-- MELLON_secure_key
-- MELLON_username
-- MELLON_email
+- MELLON_secure_key (required)
```
Comment (sumerjabri) [inline source/site-administrators/engine/engine-headers-based-authentication.rst:68]: We should reconsider this default prefix in code and here. Once we decide on the code, we can adjust this.

File: source/site-administrators/engine/engine-headers-based-authentication.rst [hunk 4]
```diff
@@ -134,27 +175,34 @@ To add optional attributes, add the following inside the ``<headers>`` tag:
...
-Open the Engine ``site-config.xml`` file in Studio, by navigating from the ``Sidebar`` to ``Site Config`` > ``Configuration``, and finally picking up the ``Engine Site Configuration`` option from the dropdown.
+Open the Engine ``site-config.xml`` file in Studio, by navigating from the ``Sidebar`` to
+``Site Config`` > ``Configuration``, and finally picking up the ``Engine Site Configuration`` option from the dropdown.
 
-You can also access the ``site-config.xml`` using your favorite editor under ``CRAFTER_HOME/data/repos/sites/SITENAME/sandbox/config/engine/site-config.xml``
+You can also access the ``site-config.xml`` using your favorite editor under
+``CRAFTER_HOME/data/repos/sites/SITENAME/sandbox/config/engine/site-config.xml``
 
-Add the following, where we are enabling Engine headers authentication and requiring authentication for all urls in the site in addition to changing the default value for the header attribute ``MELLON_secure_key`` to ``my_updated_token``. :
+Add the following, where we are enabling Engine headers authentication and requiring authentication for all urls in the
```
Comment (sumerjabri) [inline source/site-administrators/engine/engine-headers-based-authentication.rst:204]: URLs

**Review by sumerjabri — CHANGES_REQUESTED** [review]
See notes. Also, hold on merge until we decide the default prefix.

**Review by sumerjabri — CHANGES_REQUESTED** [review]
Please address the notes.

**Review by alhambrav — APPROVED** [review]

**Review by sumerjabri — APPROVED** [review]

**Pull request merged**